### PR TITLE
[fix](source) fix  projection and function pushdown not functioning correctly

### PR DESCRIPTION
**Pull request by vinlee19**
# Proposed changes

Issue [376](https://github.com/apache/doris-flink-connector/issues/376): close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicTableSource.java [hunk 1]
```diff
@@ -212,5 +200,17 @@ public boolean supportsNestedProjection() {
...
+            this.readOptions.setFilterQuery(filterQuery);
+        }
+        if (StringUtils.isNullOrWhitespaceOnly(readOptions.getReadFields())) {
+            String[] selectFields =
+                    DataType.getFieldNames(physicalRowDataType).toArray(new String[0]);
+            this.readOptions.setReadFields(
+                    Arrays.stream(selectFields)
+                            .map(item -> String.format("`%s`", item.trim().replace("`", "")))
+                            .collect(Collectors.joining(", ")));
+        }
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicTableSource.java:214]: What are the benefits of putting it below?

Comment (vinlee19) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicTableSource.java]: The method `getScanRuntimeProvider` is executed before `applyProjection`. Therefore, initializing `readFields` in `getScanRuntimeProvider` will result in querying all fields in Doris, leading to type conversion errors.

File: flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicTableSource.java [hunk 2]
```diff
@@ -194,7 +185,8 @@ public Result applyFilters(List<ResolvedExpression> filters) {
         DorisExpressionVisitor expressionVisitor = new DorisExpressionVisitor();
         for (ResolvedExpression filter : filters) {
             String filterQuery = filter.accept(expressionVisitor);
-            if (!StringUtils.isNullOrWhitespaceOnly(filterQuery)) {
+            if (StringUtils.isNullOrWhitespaceOnly(filterQuery)
+                    && !StringUtils.isNullOrWhitespaceOnly(filterQuery)) {
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicTableSource.java:189]: This is always false？

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**